### PR TITLE
Test/e2 e ci

### LINKED ISSUE #57: Add E2E tests to GA
Add tests to the CI pipeline so tests are run on every push to PR

### PR DESCRIPTION
**Pull request by jtsang586**
Run Cypress tests in CI

Closes #57

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  cypress-run:
+    name: Run end-to-end tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
```
Comment (dexturr) [inline .github/workflows/cypress.yml:22]: chore (blocking): this ref will not exist if this is not a pull request but the tirggers are also on push. You will need to add a separate job for on push

Comment (dexturr) [reply .github/workflows/cypress.yml:22]: For push I suggest using the same approach they use here: https://nx.dev/ci/monorepo-ci-github-actions defining 2 steps one for push (lines: 13-24) and one for pull request (line 26) you have the PR one but we will need the other as well to ensure that nx affected also works properly on push

**Review by dexturr — CHANGES_REQUESTED** [review]
Great stuff in general dude :+1: nicely done, one small edge case which is hard to miss though push events will cause this to fail, have put some suggestion on how to fix this down 👍 

If you need to test the push flow you can always change the trigger to work on push of your branch only while testing the CI.

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (jtsang586) [reply .github/workflows/cypress.yml:41]: Done!

**Review by dexturr — APPROVED** [review]
